### PR TITLE
Css tweaks and icon change for sidebar

### PR DESCRIPTION
**Pull request by aguscruiz**
Resolves https://github.com/brave/brave-browser/issues/42068

New sidebar expand/collapse
image

image

Additional changes:
- Add transition to conversation list items on hover

- Tweak background color for hover
- Add spacing between conversation list items.

image

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/ai_chat/resources/page/components/full_page/index.tsx
Comment (aguscruiz) [inline components/ai_chat/resources/page/components/full_page/index.tsx:1]: @fallaciousreasoning @petemill I tried making the icon change here when the sidebar is expanded/collapsed but it doesn't seem to work :( maybe you can give me a hand here

Comment (fallaciousreasoning) [reply components/ai_chat/resources/page/components/full_page/index.tsx:1]: Hey yeah, this code is pretty hard to grok - definitely took me a while 

try using 

```
asideAnimationRef.current?.playbackRate === 1 ? 'sidepanel-retract' : 'sidepanel-open'
```

Comment (aguscruiz) [reply components/ai_chat/resources/page/components/full_page/index.tsx:1]: That didn't work for me either :(

Comment (aguscruiz) [reply components/ai_chat/resources/page/components/full_page/index.tsx:1]: Both the original and this suggestion seem to change the icon, but for some reason it just doesn't show it on the ui


image

image

Comment (fallaciousreasoning) [reply components/ai_chat/resources/page/components/full_page/index.tsx:1]: Ah yeah, you probably need to add the new icon to the `leo_icons` array in `brave/ui/webui/resources/BUILD.gn`

**Comment by fallaciousreasoning** [comment]
New icons should be in brave-core if you rebase onto latest master @aguscruiz!

**Comment by aguscruiz** [comment]
> New icons should be in brave-core if you rebase onto latest master @aguscruiz!

Awesome, thanks!

File: package-lock.json [hunk 1]
```diff
@@ -753,7 +753,8 @@
     "node_modules/@brave/leo": {
       "version": "0.0.1",
       "resolved": "git+ssh://git@github.com/brave/leo.git#349a774b7cd5151db14dae6144f7e47168fe7432",
-      "integrity": "sha512-1iQ+4+ZoYlJm4RfeEUzuSoopZjzat0iIZNc0wH7wpo5iHFmcX03ptb5onu5Ck6UHB0SMR2dfOin4n9j46stM+A==",
```
Comment (fallaciousreasoning) [inline package-lock.json:756]: @aguscruiz you should revert this file - its not really part of your change:

```
git checkout origin/master -- package-lock.json
```

Comment (aguscruiz) [reply package-lock.json]: Done!

File: components/ai_chat/resources/page/components/full_page/index.tsx [hunk 2]
```diff
@@ -90,13 +90,13 @@ export default function FullScreen() {
     <div className={styles.fullscreen}>
       <div className={styles.left}>
         <div className={styles.controls}>
-          <Button
+            <Button
```
Comment (fallaciousreasoning) [inline components/ai_chat/resources/page/components/full_page/index.tsx:93]: nit: I think this should be indented two spaces instead of 4

**Review by fallaciousreasoning — APPROVED** [review]
lgtm with a nit

**Pull request merged**